http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/RuntimeRoutes.scala

```diff
@@ -0,0 +1,145 @@
+    implicit val traceId = ApplicativeAsk.const[IO, TraceId](TraceId(UUID.randomUUID()))
+    // https://doc.akka.io//docs/akka-http/current/routing-dsl/path-matchers.html?language=scala#examples
+    pathPrefix("google" / "v1" / "runtime") {
+      pathPrefix(googleProjectSegment / Segment) { (googleProject, runtimeNameString) =>
+        validateRuntimeNameDirective(runtimeNameString) { runtimeName =>
+          pathEndOrSingleSlash {
+            post {
+              entity(as[CreateRuntime2Request]) { req =>
+                complete(
+                  createRuntimeHandler(
+                    userInfo,
+                    googleProject,
+                    runtimeName,
+                    req
+                  )
+                )
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private[api] def createRuntimeHandler(userInfo: UserInfo,
+                                        googleProject: GoogleProject,
+                                        runtimeName: RuntimeName, //TODO: rename this to RuntimeName once Rob's PR is in
+                                        req: CreateRuntime2Request): IO[ToResponseMarshallable] =
+    for {
+      traceId <- IO(UUID.randomUUID())
+      implicit0(context: ApplicativeAsk[IO, RuntimeServiceContext]) <- timer.clock
+        .realTime(TimeUnit.MILLISECONDS)
+        .map(
+          n =>
+            ApplicativeAsk
+              .const[IO, RuntimeServiceContext](RuntimeServiceContext(TraceId(traceId), Instant.ofEpochMilli(n)))
+        )
+      _ <- runtimeService.createRuntime(
+        userInfo,
+        googleProject,
+        runtimeName,
+        req
+      )
+    } yield StatusCodes.Accepted
+}
+
+object RuntimeRoutes {
+  implicit val gceConfigDecoder: Decoder[RuntimeConfigRequest.GceConfig] = Decoder.forProduct2(
+    "machineType",
+    "diskSize"
+  )((mt, ds) => RuntimeConfigRequest.GceConfig(mt, ds))
+
+  implicit val runtimeConfigDecoder: Decoder[RuntimeConfigRequest] = Decoder.instance { x =>
+    //For newer version of requests, we use `cloudService` field to distinguish whether user is
+    for {
+      cloudService <- x.downField("cloudService").as[CloudService]
+      r <- cloudService match {
+        case CloudService.Dataproc =>
+          x.as[RuntimeConfigRequest.DataprocConfig]
+        case CloudService.GCE =>
+          x.as[RuntimeConfigRequest.GceConfig]
+      }
+    } yield r
+  }
+
+  implicit val createRuntimeRequestDecoder: Decoder[CreateRuntime2Request] = Decoder.instance { c =>
+    for {
+      l <- c.downField("labels").as[Option[LabelMap]]
+      je <- c.downField("jupyterExtensionUri").as[Option[GcsPath]]
+      jus <- c.downField("jupyterUserScriptUri").as[Option[UserScriptPath]]
+      jsus <- c.downField("jupyterStartUserScriptUri").as[Option[UserScriptPath]]
+      rc <- c.downField("runtimeConfig").as[Option[RuntimeConfigRequest]]
+      p <- c.downField("properties").as[Option[LabelMap]]
```

Comment:
You mentioned putting `properties` inside of `DataprocConfig` -- that's probably a good idea